### ISSUE TITLE
tune() `alpha` and `lambda` hyperparameters in XGBoost

### ISSUE THREAD

**Issue by driapitek** [tidymodels/dials#176]
I want to be able to `tune()` all available hyperparameters in `xgboost` (in particular `lambda` and `alpha`). 
It seems that I can only `tune()` what is listed in the help [https://parsnip.tidymodels.org/reference/details_boost_tree_xgboost.html](https://parsnip.tidymodels.org/reference/details_boost_tree_xgboost.html):
- `tree_depth`
- `trees`
- `learn_rate`
- `mtry`
- `min_n`
- `loss_reduction`
- `sample_size`
- `stop_iter`

Currently, to specify `lambda` I need to specify explicitly:
```r
boost_tree("regression",
    trees = tune(),
    learn_rate = tune()) %>%
set_engine("xgboost", lambda = 100)
```

How can I `tune()` `lambda` in that case? In general, why can't I `tune()` all hyperparameters available on
[https://xgboost.readthedocs.io/en/latest/parameter.html](https://xgboost.readthedocs.io/en/latest/parameter.html)

**Comment by juliasilge** [comment]
Next step is to add the engine-specific `lambda` and `alpha` [here](https://github.com/tidymodels/dials/blob/master/R/param_engine_xgboost.R). This might be a good opportunity for a first-time contributor.

**Comment by hfrick** [comment]
I think `penalty_L1()` and `penalty_L2()` would be good! This would be nice to put in the next release for dials which we are aiming for for next week. @joeycouse  would that timeline work for you or would you prefer us adding it in ourselves?

**Comment by joeycouse** [comment]
I'll try my best! I just opened a PR but I think some edits will need to be done on the parsnip side.

**Comment by hfrick** [comment]
Thank you @joeycouse !

**Comment by hfrick** [comment]
Closed in #179
